pages/latest/[urlTitle].tsx

```diff
@@ -41,20 +45,57 @@ const componentsUI: MDXProviderComponentsProp = {
     a: LinkMarkdown
 };
 
-const UrlTitle = ({ data, source, error }: PageData): JSX.Element => {
+const UrlTitle = ({ data, source, toc, error }: PageData): JSX.Element => {
     const content = source ? hydrate(source, { components: componentsUI }) : null;
+    // ---- Table Of Content Active Item
+    const [tocActive, setTocActive] = useState(null);
+    const observer = useRef(null);
```

Comment:
What this is doing here with the `useRef`?

Reply:
To the `useRef` we are assigning the intersection observer container.

Reply:
@fmontes @rjvelazco  Careful when using `useRef` because if `.current` changes it won't trigger a re-render. Is that what you want?